akka-http/src/main/scala/akka/http/scaladsl/server/directives/MdcLoggingDirectives.scala

```diff
@@ -0,0 +1,57 @@
+package akka.http.scaladsl.server.directives
+
+import akka.event.{
+  DefaultLoggingFilter,
+  DiagnosticMarkerBusLoggingAdapter,
+  LogSource,
+  MarkerLoggingAdapter
+}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+
+/**
+ * @groupname mdc Mdc logging directives
+ * @groupprio mdc 240
+ */
+trait MdcLoggingDirectives {
+
+  import MdcLoggingDirectives.extractDiagnosticMarkerLog
+
+  def withMdcLogging: Directive0 =
+    extractDiagnosticMarkerLog
+      .flatMap { dmbla: DiagnosticMarkerBusLoggingAdapter =>
+        mapRequestContext { ctx =>
+          ctx.withLog(dmbla)
+        }
+      }
+
+  def extractMarkerLog: Directive1[MarkerLoggingAdapter] =
```

Comment:
Do we need that, aren't all logger also `MarkerLoggingAdapter`s anyway even without `withMdcLogging` enabled?

Reply:
>  aren't all logger also MarkerLoggingAdapters anyway even without withMdcLogging enabled?

I don't think so. I'm assuming you're implying we could just use `extractLog` instead of having this.

`extractLog` returns a `LoggingAdapter`: https://github.com/akka/akka-http/blob/90b5ecbc6f41b13af02153f3f61accb0458cf88e/akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala#L279

`MarkerLoggingAdapter` has a few extra methods, e.g., https://github.com/akka/akka/blob/main/akka-actor/src/main/scala/akka/event/Logging.scala#L1848-L1849

Maybe it's true that the `LoggingAdapter` that gets returned from `extractLog` happens to always also extend `MarkerLoggingAdapter`, but that's not specified in the return type, so we can't access those methods.